### PR TITLE
Added Swift and Kotlin searches

### PR DESCRIPTION
**Pull request by Fedorsturov**
Issue #3 
Added Swift and Kotlin searches

### CODE REVIEW

**Comment by dharan1011** [comment]
Great job 👍 . But as mentioned the Document the code so that a novice programmer can understand whats happening in the code.

**Comment by Fedorsturov** [comment]
@dharan1011 I have added docstrings to these examples

File: Swift/BinarySearch.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+func binarySearch<T: Comparable>(xs: [T], x: T) -> Int? {
+  var recurse: ((Int, Int) -> Int?)!
+  recurse = {(low, high) in switch (low + high) / 2 {
+    case _ where high < low: return nil
+    case let mid where xs[mid] > x: return recurse(low, mid - 1)
+    case let mid where xs[mid] < x: return recurse(mid + 1, high)
+    case let mid: return mid
+  }}
+  return recurse(0, xs.count - 1)
+}
```
Comment (dharan1011) [inline Swift/BinarySearch.swift]: Great Job 👍

Comment (dharan1011) [reply Swift/BinarySearch.swift]: But still closure can be bit confusing for a novice developer try simplifying what you have implemented.

File: Swift/LinearSearch.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//Swift has indexOf for linear search
+let haystack = ["Zig","Zag","Wally","Ronald","Bush","Krusty","Charlie","Bush","Bozo"]
+for needle in ["Washington","Bush"] {
```
Comment (dharan1011) [inline Swift/LinearSearch.swift]: `["Washington","Bush"]` could have been extracted to variable

Comment (dharan1011) [reply Swift/LinearSearch.swift]: The way you implemented is very clear and crisp. 👍

File: Kotlin/LinearSearch.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    println(haystack)
+    var needle = "Zag"
+    var index  = haystack.indexOf(needle)
+    val index2 = haystack.lastIndexOf(needle)
+    println("\n'$needle' first occurs at index $index of the list")
+    println("'$needle' last  occurs at index $index2 of the list\n")
+    needle = "Donald"
+    index  = haystack.indexOf(needle)
+    if (index == -1) throw Exception("$needle does not occur in the list")
+}
```
Comment (dharan1011) [inline Kotlin/LinearSearch.kt]: This isn't a proper implementation of linear search. Implement it like what you have in the Swift.

File: Kotlin/BinarySearch.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// Kotlin implementation of recursive Binary Search
```
Comment (dharan1011) [inline Kotlin/BinarySearch.kt]: Great Job 👍 Documenting the code.

File: Kotlin/BinarySearch.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Set m (the position of the middle element) to the floor (the largest previous integer) of (L + R) / 2.
+If Am < T, set L to m + 1 and go to step 2.
+If Am > T, set R to m − 1 and go to step 2.
+Now Am = T, the search is done; return m.
+
+NB: We add binary search capabilities to arrays.
+
+*/
+fun <T : Comparable<T>> Array<T>.iterativeBinarySearch(target: T): Int {
+    var hi = size - 1
```
Comment (dharan1011) [inline Kotlin/BinarySearch.kt]: the variable`hi` name could have been clear i.e high and `lo` could be named low

File: Kotlin/BinarySearch.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        else if (this[guess] < target) lo = guess + 1
+        else return guess
+    }
+    return -1
+}
+
+fun <T : Comparable<T>> Array<T>.recursiveBinarySearch(target: T, lo: Int, hi: Int): Int {
+    if (hi < lo) return -1
+
+    val guess = (hi + lo) / 2
```
Comment (dharan1011) [inline Kotlin/BinarySearch.kt]: rename guess to mid for better understanding

**Pull request merged**